### PR TITLE
Fix deleteBatchesForIds when multiple selectionArgs

### PR DESCRIPTION
**Pull request by danybony**
### Problem
`BatchRepository.deleteBatchesForIds` was working just because it was always invoked with a single element in the `batchIdsToDelete` parameter list.

### Solution
We updated the notils dependency to 2.2.13 and use the new QueryUtils method in order to create the right placeholder string, depending on the number of selection args.

We also replace the duplicated code in `DownloadsRepository`

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/182/
Test FAILed.

**Comment by xrigau** [comment]
oh and nao we have to review more code :sob:

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java [hunk 1]
```diff
@@ -198,9 +199,9 @@ private void deleteBatchesForIds(List<Long> batchIdsToDelete, Collection<FileDow
             }
         }
 
-        String selection = StringUtils.join(batchIdsToDelete, ", ");
-        String[] selectionArgs = {selection};
-        resolver.delete(downloadsUriProvider.getBatchesUri(), DownloadContract.Batches._ID + " IN (?)", selectionArgs);
+        String where = DownloadContract.Batches._ID + " IN (" + QueryUtils.createSelectionPlaceholdersOfSize(batchIdsToDelete.size()) + ")";
```
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: would be nice to extract some vars here

Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: :+1:

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/183/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java [hunk 2]
```diff
@@ -410,6 +410,7 @@ public DownloadManager(Context context, ContentResolver contentResolver, boolean
         this.systemFacade = systemFacade;
         GlobalState.setContext(context);
         GlobalState.setVerboseLogging(verboseLogging);
+        Log.setShowLogs(verboseLogging);
```
Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: This is needed singe NoTils 2.2.12 :/

**Comment by zegnus** [comment]
:+1: LGTM, +10 minutes for more comments

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java (hunk 2 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: so this'll be removed in a later PR?

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: @blundell ^?

Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: Yes, this will be removed when we'll get rid of the Notils logger

Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: Better fix notils logger.

Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: changing this on my branch, so you can carry on

**Pull request merged**